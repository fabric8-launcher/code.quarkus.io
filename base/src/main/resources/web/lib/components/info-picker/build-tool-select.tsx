import * as React from 'react';
import { useAnalyticsEditionField } from '../../core/analytics';
import { InputProps } from '../../core/types';
import { FaAngleDown } from 'react-icons/fa';

export const BuildToolSelect = (props: InputProps<string>) => {
  const onChangeWithDirty = useAnalyticsEditionField('buildTool', props.onChange)[1];
  const adaptedOnChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
    onChangeWithDirty(e.currentTarget.value);
  };
  return (
    <div className="form-group form-group-select">
      <label className="form-group-label" htmlFor="buildtool" arial-label="Choose build tool"><span
        className="form-group-label-text">Build Tool</span></label>
      <select id="buildtool" value={props.value} aria-label="Edit build tool" onChange={adaptedOnChange} className={'form-group-control form-group-select'}>
        <option value={'MAVEN'}>Maven</option>
        <option value={'GRADLE'}>Gradle</option>
        <option value={'GRADLE_KOTLIN_DSL'}>Gradle with Kotlin DSL</option>
      </select>
      <FaAngleDown />
    </div>
  );
};