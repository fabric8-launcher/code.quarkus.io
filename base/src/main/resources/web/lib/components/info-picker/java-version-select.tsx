import * as React from 'react';
import { useAnalyticsEditionField } from '../../core/analytics';
import { InputProps } from '../../core/types';
import { FaAngleDown } from 'react-icons/fa';
import { JavaCompatibility } from '../api/model';

interface JavaVersionSelectProps extends InputProps<string> {
  javaCompatibility?: JavaCompatibility;
}

export const JavaVersionSelect = (props: JavaVersionSelectProps) => {
  const onChangeWithDirty = useAnalyticsEditionField('buildTool', props.onChange)[1];
  const adaptedOnChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
    onChangeWithDirty(e.currentTarget.value);
  };
  return (
    <div className="form-group form-group-select javaversion-select">
      <label className="form-group-label" htmlFor="javaversion" arial-label="Choose Java Version"><span
        className="form-group-label-text">Java Version</span></label>
      <select id="javaversion" aria-label="Edit Java version" value={props.value || props.javaCompatibility?.recommended || '17'} onChange={adaptedOnChange} className={'form-group-control form-group-select'}>
        {props.javaCompatibility?.versions.map((version) => <option key={version} value={version}>{version}</option>)}
      </select>
      <FaAngleDown />
    </div>
  );
};