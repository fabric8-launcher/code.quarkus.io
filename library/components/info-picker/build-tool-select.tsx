import * as React from 'react';
import { useAnalyticsEditionField } from '@quarkusio/code-quarkus.core.analytics';
import { InputProps } from '@quarkusio/code-quarkus.core.types';
import { FaAngleDown } from 'react-icons/fa';

export const BuildToolSelect = (props: InputProps<string>) => {
  const onChangeWithDirty = useAnalyticsEditionField('buildTool', props.onChange)[1];
  const adaptedOnChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
    onChangeWithDirty(e.currentTarget.value);
  };
  return (
    <div className="form-group form-group-select">
      <label className="form-group-label" htmlFor="buildtool" arial-label="Choose build tool"><span
        className="form-group-label-text">Build Tool</span></label>
      <select id="buildtool" value={props.value} onChange={adaptedOnChange} className={'form-group-control form-group-select'}>
        <option value={'MAVEN'}>Maven</option>
        <option value={'GRADLE'}>Gradle</option>
        <option value={'GRADLE_KOTLIN_DSL'}>Gradle with Kotlin DSL</option>
      </select>
      <FaAngleDown />
    </div>
  );
};