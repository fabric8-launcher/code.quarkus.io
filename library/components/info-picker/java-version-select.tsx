import * as React from 'react';
import { useAnalyticsEditionField } from '@quarkusio/code-quarkus.core.analytics';
import { InputProps } from '@quarkusio/code-quarkus.core.types';
import { FaAngleDown } from 'react-icons/fa';

export const JavaVersionSelect = (props: InputProps<string>) => {
  const onChangeWithDirty = useAnalyticsEditionField('buildTool', props.onChange)[1];
  const adaptedOnChange = (e: React.ChangeEvent<HTMLSelectElement>) => {
    onChangeWithDirty(e.currentTarget.value);
  };
  return (
    <div className="form-group form-group-select javaversion-select">
      <label className="form-group-label" htmlFor="javaversion" arial-label="Choose Java Version"><span
        className="form-group-label-text">Java Version</span></label>
      <select id="javaversion" value={props.value} onChange={adaptedOnChange} className={'form-group-control form-group-select'}>
        <option value={'11'}>11</option>
        <option value={'17'}>17</option>
      </select>
      <FaAngleDown />
    </div>
  );
};