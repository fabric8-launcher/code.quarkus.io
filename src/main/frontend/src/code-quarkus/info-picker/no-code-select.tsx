import { InputProps, useAnalyticsEditionField } from '../../core';
import React, { ChangeEvent } from 'react';
import { FormGroup, Tooltip } from '@patternfly/react-core';


export const NoCodeSelect = (props: InputProps<boolean>) => {
    const onChangeWithDirty = useAnalyticsEditionField('no-code', props.onChange)[1];
    const adaptedOnChange = (e: ChangeEvent<HTMLSelectElement>) => {
        onChangeWithDirty(e.target.value === 'true', e);
    };
    return (
        <Tooltip
            position="right"
            content={<span>The flag <span className="codestart-icon" /> means the extension provides starter code. You can also choose to have an empty Quarkus project.</span>}
            exitDelay={0}
            zIndex={200}
        >
            <FormGroup
                fieldId="no-code"
                label={<span><span className="codestart-icon" />Starter Code</span>}
                aria-label="Starter Code">
                <select id="no-code" value={props.value ? 'true' : 'false'} onChange={adaptedOnChange} className={'pf-c-form-control'}>
                    <option value={'false'}>Yes</option>
                    <option value={'true'}>No</option>
                </select>
            </FormGroup>
        </Tooltip>
    );
};
