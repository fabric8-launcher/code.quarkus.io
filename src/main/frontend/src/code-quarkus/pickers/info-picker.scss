.info-picker {
  
  &.horizontal {
    display: flex;

    .base-settings {
      order: 1;
      flex-grow: 1;
      padding: 0 7px;
      align-self: flex-start;
      margin: 0;
    }

    .toggle-panel {
      align-self: flex-start;
      order: 4;
      flex-grow: 1;
      padding: 0 7x;
     
    }

    .toggle-panel.open {
      order: 2;
    }

    .toggle-button {
      flex-shrink: 1;
      order: 3;
      align-self: flex-end;
      padding: 0 10px;
    }
    
  }
}