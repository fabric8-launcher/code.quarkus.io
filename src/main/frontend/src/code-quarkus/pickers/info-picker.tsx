import React, {ChangeEvent, useEffect} from 'react';
import {ExtendedTextInput, InputPropsWithValidation, optionalBool, TogglePanel} from '../../core';
import './info-picker.scss';
import {FormGroup} from "@patternfly/react-core";

export interface InfoPickerValue {
  groupId?: string;
  artifactId?: string;
  version?: string;
  packageName?: string;
  buildTool?: string;
}

interface InfoPickerProps extends InputPropsWithValidation<InfoPickerValue> {
  showMoreOptions?: boolean;
}

const ARTIFACTID_PATTERN = /^[a-z][a-z0-9-._]*$/;
const GROUPID_PATTERN = /^([a-zA-Z_$][a-zA-Z\d_$]*\.)*[a-zA-Z_$][a-zA-Z\d_$]*$/;

const isValidId = (value?: string) => !!value && ARTIFACTID_PATTERN.test(value || '');
const isValidGroupId = (value?: string) => !!value && GROUPID_PATTERN.test(value);
const isValidInfo = (value: InfoPickerValue) => {
  return isValidGroupId(value.groupId)
    && isValidId(value.artifactId)
    && !!value.version
    && (!value.packageName || isValidGroupId(value.packageName))
}

export const InfoPicker = (props: InfoPickerProps) => {
  const { value, isValid, onChange } = props;
  const onInputChange = (value: InfoPickerValue) => {
    props.onChange(value, isValidInfo(value));
  };

  useEffect(() => {
    if (isValid !== isValidInfo(value)) {
      onChange(value, !isValid);
    }
  }, [value, isValid, onChange])

  const onGroupIdChange = (newValue: string) => onInputChange({ ...value, groupId: newValue });
  const onArtifactIdChange = (newValue: string) => onInputChange({ ...value, artifactId: newValue });
  const onVersionChange = (newValue: string) => onInputChange({ ...value, version: newValue });
  const onPackageNameChange = (newValue: string) => onInputChange({ ...value, packageName: newValue });
  const onBuildToolChange = (event: ChangeEvent<HTMLSelectElement>) => onInputChange({ ...value, buildTool: event.target.value });

  return (
    <div className={`info-picker horizontal`}>
      <div className="base-settings pf-c-form">
        <ExtendedTextInput
          label="Group"
          isRequired
          type="text"
          id="groupId"
          name="groupId"
          aria-label="Edit groupId"
          value={value.groupId || ''}
          autoComplete="off"
          onChange={onGroupIdChange}
          pattern={GROUPID_PATTERN.source}
          isValid={isValidGroupId(value.groupId)}
        />
        <ExtendedTextInput
          label="Artifact"
          isRequired
          type="text"
          id="artifactId"
          name="artifactId"
          aria-label="Edit artifactId"
          value={value.artifactId || ''}
          autoComplete="off"
          onChange={onArtifactIdChange}
          pattern={ARTIFACTID_PATTERN.source}
          isValid={isValidId(value.artifactId)}
        />
      </div>
      {optionalBool(props.showMoreOptions, true) && (
        <TogglePanel id="info-extended" mode="horizontal" openLabel="Configure more options">
          <div className="extended-settings pf-c-form">
            <ExtendedTextInput
              label="Version"
              isRequired
              type="text"
              id="version"
              name="version"
              aria-label="Edit version"
              value={value.version || ''}
              autoComplete="off"
              onChange={onVersionChange}
              isValid={!!value.version}
            />
            <ExtendedTextInput
              label="Package Name"
              isRequired
              type="text"
              id="packageName"
              name="packageName"
              aria-label="Edit package name"
              value={value.packageName || value.groupId || ''}
              autoComplete="off"
              onChange={onPackageNameChange}
              pattern={GROUPID_PATTERN.source}
              isValid={isValidGroupId(value.packageName || value.groupId)}
            />
            <FormGroup
                fieldId="buildTool"
                label="Build Tool"
                aria-label="Choose build tool">
              <select id="buildtool" value={value.buildTool} onChange={onBuildToolChange} className={'pf-c-form-control'}>
                <option value={"MAVEN"}>Maven</option>
                <option value={"GRADLE"}>Gradle</option>
              </select>
            </FormGroup>
          </div>
        </TogglePanel>
      )}
    </div>
  );
};
