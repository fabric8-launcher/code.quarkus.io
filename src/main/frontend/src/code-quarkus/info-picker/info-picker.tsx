import React from 'react';
import { ExtendedTextInput, InputProps, optionalBool, TogglePanel } from '../../core';
import './info-picker.scss';
import { BuildToolSelect } from './build-tool-select';
import { NoCodeSelect } from './no-code-select';

export interface InfoPickerValue {
  groupId?: string;
  artifactId?: string;
  version?: string;
  noCode?: boolean;
  buildTool?: string;
}

interface InfoPickerProps extends InputProps<InfoPickerValue> {
  showMoreOptions?: boolean;
  quarkusVersion: string;
}

const ARTIFACTID_PATTERN = /^[a-z][a-z0-9-._]*$/;
const GROUPID_PATTERN = /^([a-zA-Z_$][a-zA-Z\d_$]*\.)*[a-zA-Z_$][a-zA-Z\d_$]*$/;

const isValidId = (value?: string) => !!value && ARTIFACTID_PATTERN.test(value || '');
const isValidGroupId = (value?: string) => !!value && GROUPID_PATTERN.test(value);
export const isValidInfo = (value: InfoPickerValue) => {
  return isValidGroupId(value.groupId)
    && isValidId(value.artifactId)
    && !!value.version;
};

export const InfoPicker = (props: InfoPickerProps) => {

  const onInputChange = props.onChange;

  const onGroupIdChange = (newValue: string) => onInputChange({ ...props.value, groupId: newValue });
  const onArtifactIdChange = (newValue: string) => onInputChange({ ...props.value, artifactId: newValue });
  const onVersionChange = (newValue: string) => onInputChange({ ...props.value, version: newValue });
  const onNoCodeChange = (newValue: boolean) => onInputChange({ ...props.value, noCode: newValue });
  const onBuildToolChange = (newValue: string) => onInputChange({ ...props.value, buildTool: newValue });

  return (
    <div className={`info-picker horizontal`}>
      <div className="base-settings pf-c-form">
        <ExtendedTextInput
          label="Group"
          isRequired
          type="text"
          id="groupId"
          name="groupId"
          aria-label="Edit groupId"
          value={props.value.groupId || ''}
          autoComplete="off"
          onChange={onGroupIdChange}
          pattern={GROUPID_PATTERN.source}
          isValid={isValidGroupId(props.value.groupId)}
        />
        <ExtendedTextInput
          label="Artifact"
          isRequired
          type="text"
          id="artifactId"
          name="artifactId"
          aria-label="Edit artifactId"
          value={props.value.artifactId || ''}
          autoComplete="off"
          onChange={onArtifactIdChange}
          pattern={ARTIFACTID_PATTERN.source}
          isValid={isValidId(props.value.artifactId)}
        />
        <BuildToolSelect onChange={onBuildToolChange} value={props.value.buildTool || 'MAVEN'}/>
      </div>
      {optionalBool(props.showMoreOptions, true) && (
        <TogglePanel id="info-extended" mode="horizontal" openLabel="Configure more options" event={['UX', 'Application Info - Configure More Options']}>
          <div className="extended-settings pf-c-form">
            <ExtendedTextInput
              label="Version"
              isRequired
              type="text"
              id="projectVersion"
              name="projectVersion"
              aria-label="Edit project version"
              value={props.value.version || ''}
              autoComplete="off"
              onChange={onVersionChange}
              isValid={!!props.value.version}
            />
            <NoCodeSelect onChange={onNoCodeChange} value={props.value.noCode || false} />
          </div>
        </TogglePanel>
      )}
    </div>
  );
};
