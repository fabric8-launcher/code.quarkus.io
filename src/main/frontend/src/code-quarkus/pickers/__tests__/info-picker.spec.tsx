import React from 'react';
import {cleanup, fireEvent, render, RenderResult} from '@testing-library/react';
import {InfoPicker} from '../info-picker';
import {act} from 'react-dom/test-utils';

jest.useFakeTimers();

afterEach(cleanup);

describe('<InfoPicker />', () => {
  it('renders the InfoPicker correctly', () => {
    const onChange = jest.fn();
    const comp = render(<InfoPicker value={{ groupId: 'org.test', version: '1.0.0', artifactId: 'test' }} isValid={true} onChange={() => { }} quarkusVersion="1.0.0"/>);
    expect(comp.asFragment()).toMatchSnapshot();
    expect(onChange).toBeCalledTimes(0);
  });

  it('receive isValid after init when it has change', () => {
    const onChangeMock = jest.fn();
    const value = { groupId: 'org.test', version: '1.0.0', artifactId: 'test', packageName: 'org.package' };
    let comp: RenderResult;
    act(() => {
      comp = render(<InfoPicker value={value} isValid={false} onChange={onChangeMock} quarkusVersion="1.0.0"/>);
    });
    act(() => { jest.runTimersToTime(210); });
    expect(onChangeMock).lastCalledWith(value, true);
  });

  it('receive isValid false when using invalid groupId', () => {
    const onChangeMock = jest.fn();
    const value = { groupId: 'org.test', version: '1.0.0', artifactId: 'test', packageName: 'org.package' };
    let comp: RenderResult;
    act(() => {
      comp = render(<InfoPicker value={value} isValid={true} onChange={onChangeMock} quarkusVersion="1.0.0"/>);
    });
    fireEvent.change(comp!.getByLabelText('Edit groupId'), { target: { value: 'com.' } });
    act(() => { jest.runTimersToTime(210); });
    expect(onChangeMock).lastCalledWith({...value, groupId: 'com.'}, false);
  });

  it('receive isValid false when using invalid artifactId', () => {
    const onChangeMock = jest.fn();
    const value = { groupId: 'org.test', version: '1.0.0', artifactId: 'test', packageName: 'org.package' };
    let comp: RenderResult;
    act(() => {
      comp = render(<InfoPicker value={value} isValid={true} onChange={onChangeMock} quarkusVersion="1.0.0"/>);
    });
    fireEvent.change(comp!.getByLabelText('Edit artifactId'), { target: { value: 'invalid id' } });
    act(() => { jest.runTimersToTime(210); });
    expect(onChangeMock).lastCalledWith({...value, artifactId: 'invalid id'}, false);
  });

  it('receive isValid false when using invalid project version', () => {
    const onChangeMock = jest.fn();
    const value = { groupId: 'org.test', version: '1.0.0', artifactId: 'test', packageName: 'org.package' };
    let comp: RenderResult;
    act(() => {
      comp = render(<InfoPicker value={value} isValid={true} onChange={onChangeMock} quarkusVersion="1.0.0"/>);
    });
    fireEvent.change(comp!.getByLabelText('Edit package name'), { target: { value: 'com.1a' } });
    act(() => { jest.runTimersToTime(210); });
    expect(onChangeMock).lastCalledWith({...value, packageName: 'com.1a'}, false);
  });

  it('receive isValid false when using invalid package name', () => {
    const onChangeMock = jest.fn();
    const value = { groupId: 'org.test', version: '1.0.0', artifactId: 'test', packageName: 'org.package' };
    let comp: RenderResult;
    act(() => {
      comp = render(<InfoPicker value={value} isValid={true} onChange={onChangeMock} quarkusVersion="1.0.0"/>);
    });
    fireEvent.change(comp!.getByLabelText('Edit package name'), { target: { value: 'com.1a' } });
    act(() => { jest.runTimersToTime(210); });
    expect(onChangeMock).lastCalledWith({...value, packageName: 'com.1a'}, false);
  });

  it('auto update untouched package name when groupId is edited', () => {
    const onChangeMock = jest.fn();
    const value = { groupId: 'org.test', version: '1.0.0', artifactId: 'test' };
    let comp: RenderResult;
    act(() => {
      comp = render(<InfoPicker value={value} isValid={true} onChange={onChangeMock} quarkusVersion="1.0.0"/>);
    });
    fireEvent.change(comp!.getByLabelText('Edit groupId'), { target: { value: 'org.test.copy' } });
    act(() => {
      comp.rerender(<InfoPicker value={{...value, groupId: 'org.test.copy'}} isValid={true} onChange={onChangeMock} quarkusVersion="1.0.0"/>);
    });
    expect(comp!.getByLabelText('Edit package name').getAttribute('value')).toBe('org.test.copy')

  });

  it('display errors when using invalid values', () => {
    const onChangeMock = jest.fn();
    const initialValue = { groupId: 'com.test', version: 'version', artifactId: 'test', packageName: 'org.package' };
    const invalidValue = { groupId: 'com.1t', version: '', artifactId: 'Te', packageName: 'org.package ' };
    let comp: RenderResult;
    act(() => {
      comp = render(<InfoPicker value={initialValue} isValid={true} onChange={onChangeMock} quarkusVersion="1.0.0"/>);
    });
    fireEvent.change(comp!.getByLabelText('Edit groupId'), { target: { value: invalidValue.groupId } });
    fireEvent.change(comp!.getByLabelText('Edit artifactId'), { target: { value: invalidValue.artifactId } });
    fireEvent.change(comp!.getByLabelText('Edit project version'), { target: { value: invalidValue.version } });
    fireEvent.change(comp!.getByLabelText('Edit package name'), { target: { value: invalidValue.packageName } });

    act(() => { jest.runTimersToTime(210); });
    expect(onChangeMock).toHaveBeenCalledTimes(1);

    expect(comp!.getByLabelText('Edit groupId').getAttribute('aria-invalid')).toBe('true');
    expect(comp!.getByLabelText('Edit artifactId').getAttribute('aria-invalid')).toBe('true');
    expect(comp!.getByLabelText('Edit project version').getAttribute('aria-invalid')).toBe('true');
    expect(comp!.getByLabelText('Edit package name').getAttribute('aria-invalid')).toBe('true');
  });
});
