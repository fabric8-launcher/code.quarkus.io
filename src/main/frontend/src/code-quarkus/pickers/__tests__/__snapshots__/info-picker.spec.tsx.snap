// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<InfoPicker /> renders the InfoPicker correctly 1`] = `
<DocumentFragment>
  <div
    class="info-picker horizontal"
  >
    <div
      class="base-settings pf-c-form"
    >
      <div
        class="pf-c-form__group"
      >
        <label
          class="pf-c-form__label"
          for="groupId"
        >
          <span
            class="pf-c-form__label-text"
          >
            Group
          </span>
        </label>
        <input
          aria-invalid="false"
          aria-label="Edit groupId"
          autocomplete="off"
          class="pf-c-form-control"
          id="groupId"
          name="groupId"
          pattern="^([a-zA-Z_$][a-zA-Z\\\\d_$]*\\\\.)*[a-zA-Z_$][a-zA-Z\\\\d_$]*$"
          required=""
          type="text"
          value="org.test"
        />
      </div>
      <div
        class="pf-c-form__group pf-m-inline"
      >
        <div
          class="pf-c-form__group"
        >
          <label
            class="pf-c-form__label"
            for="artifactId"
          >
            <span
              class="pf-c-form__label-text"
            >
              Artifact
            </span>
          </label>
          <input
            aria-invalid="false"
            aria-label="Edit artifactId"
            autocomplete="off"
            class="pf-c-form-control"
            id="artifactId"
            name="artifactId"
            pattern="^[a-z][a-z0-9-._]*$"
            required=""
            type="text"
            value="test"
          />
        </div>
        <button
          class="pf-c-button pf-m-secondary pf-m-small"
          type="button"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            height="1em"
            role="img"
            style="vertical-align: -0.125em;"
            viewBox="0 64 1024 1024"
            width="1em"
          >
            <path
              d="M874.016 800.873c-92.647 92.658-220.631 149.984-362.016 149.984-282.77 0-512-229.23-512-512s229.23-512 512-512c153.639 0 291.458 67.691 385.303 174.857l-96.32 84.283c-5.595-6.377-11.401-12.615-17.458-18.667-72.523-72.528-168.955-112.473-271.525-112.473s-198.999 39.945-271.531 112.466c-72.521 72.535-112.469 168.965-112.469 271.534s39.947 198.999 112.469 271.529c72.53 72.526 168.962 112.471 271.531 112.471s198.999-39.945 271.529-112.471l154.805-61.198-64.318 151.685zM682.425 627.568c-23.333-23.333-15.424-42.425 17.575-42.425h264c51.666 0 60 9.001 60 60v264c0 33.001-19.093 40.907-42.425 17.575l-299.15-299.15z"
              transform="rotate(180 0 512) scale(-1 1)"
            />
          </svg>
        </button>
      </div>
      <div
        aria-label="Choose build tool"
        class="pf-c-form__group"
      >
        <label
          class="pf-c-form__label"
          for="buildTool"
        >
          <span
            class="pf-c-form__label-text"
          >
            Build Tool
          </span>
        </label>
        <select
          class="pf-c-form-control"
          id="buildtool"
        >
          <option
            value="MAVEN"
          >
            Maven
          </option>
          <option
            value="GRADLE"
          >
            Gradle (Preview)
          </option>
          <option
            value="GRADLE_KOTLIN_DSL"
          >
            Gradle with Kotlin DSL (Preview)
          </option>
        </select>
      </div>
    </div>
    <div
      class="toggle-panel horizontal "
    >
      <div
        class="extended-settings pf-c-form"
      >
        <div
          class="pf-c-form__group"
        >
          <label
            class="pf-c-form__label"
            for="projectVersion"
          >
            <span
              class="pf-c-form__label-text"
            >
              Version
            </span>
          </label>
          <input
            aria-invalid="false"
            aria-label="Edit project version"
            autocomplete="off"
            class="pf-c-form-control"
            id="projectVersion"
            name="projectVersion"
            required=""
            type="text"
            value="1.0.0"
          />
        </div>
        <div
          aria-label="Examples"
          class="pf-c-form__group"
        >
          <label
            class="pf-c-form__label"
            for="no-examples"
          >
            <span
              class="pf-c-form__label-text"
            >
              <span>
                <span
                  class="codestart-example-icon"
                />
                Example code
              </span>
            </span>
          </label>
          <select
            class="pf-c-form-control"
            id="no-examples"
          >
            <option
              value="false"
            >
              Yes, Please
            </option>
            <option
              value="true"
            >
              No, Thanks
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      class="toggle-button"
    >
      <a
        aria-disabled="false"
        aria-label="Toggle panel"
        class="pf-c-button pf-m-link"
      >
        <span
          class="pf-c-button__text"
        >
          <span>
            <svg
              aria-hidden="true"
              fill="currentColor"
              height="1em"
              role="img"
              style="vertical-align: -0.125em;"
              viewBox="0 0 448 512"
              width="1em"
            >
              <path
                d="M48 32h352c26.51 0 48 21.49 48 48v352c0 26.51-21.49 48-48 48H48c-26.51 0-48-21.49-48-48V80c0-26.51 21.49-48 48-48zm140.485 355.515l123.029-123.029c4.686-4.686 4.686-12.284 0-16.971l-123.029-123.03c-7.56-7.56-20.485-2.206-20.485 8.485v246.059c0 10.691 12.926 16.045 20.485 8.486z"
                transform=""
              />
            </svg>
             Configure more options
          </span>
        </span>
      </a>
    </div>
  </div>
</DocumentFragment>
`;
